File: src/dotnet/Common/Models/Agents/KnowledgeManagementAgent.cs
using FoundationaLLM.Agent.Constants;
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Agents
{
    /// <summary>
    /// The Knowledge Management agent metadata model.
    /// </summary>
    public class KnowledgeManagementAgent : AgentBase
    {
        /// <summary>
        /// The vectorization indexing profile resource path.
        /// </summary>
        [JsonPropertyName("indexing_profile")]
        public string? IndexingProfile { get; set; }
        /// <summary>
        /// The vectorization embedding profile resource path.
        /// </summary>
        [JsonPropertyName("embedding_profile")]
        public string? EmbeddingProfile { get; set; }

        /// <summary>
        /// Set default property values.
        /// </summary>
        public KnowledgeManagementAgent() =>
            Type = AgentTypes.KnowledgeManagement;
    }
}

File: src/python/PythonSDK/foundationallm/models/metadata/knowledge_management_agent.py
"""
Class Name: KnowledgeManagementAgent

Description: Encapsulates the metadata for the agent
fulfilling the orchestration request.
"""
from typing import Optional
from .agent_base import AgentBase
from foundationallm.models.agents import AgentVectorizationSettings

class KnowledgeManagementAgent(AgentBase):
    """Knowlege Management Agent metadata model."""
    vectorization: AgentVectorizationSettings = None
